### PR TITLE
Gulp Deploy Job

### PR DESCRIPTION
**Pull request by indytechcook**
@tladendo @vietnogi for your review

Uses the https://www.npmjs.com/package/gulp-awspublish npm module to push to S3.  There are also some jshint fixes which were required to get the build working.

### CODE REVIEW

File: gulp.config.js [hunk 1]
```diff
@@ -136,7 +136,18 @@ module.exports = function () {
...
+
+    /**
+     * AWS settings
+     */
+    aws: {
+      bucket: envConfig.getVal('AWS_BUCKET', 'foo'),
+      key: envConfig.getVal('AWS_KEY', 'bar'),
+      region: envConfig.getVal('AWS_REGION', 'foo'),
+      secret: envConfig.getVal('AWS_SECRET', 'bar'),
+      cdnUrl: envConfig.getVal('AWS_CDN_URL', 'foobar')
```
Comment (vietnogi) [inline gulp.config.js]: Are the build sources referencing this CDN?

**Comment by vietnogi** [comment]
Just my one comment, other stuff lgtm.

**Comment by indytechcook** [comment]
Thanks Alex. I need to add the CDN url to the injection.  Don't merge thus
yet.

**Comment by indytechcook** [comment]
@vietnogi and @tladendo a bunch of changes around the files and build.  Running "gulp optimize" works but the build isn't concatenating correctly yet.  Give it a look, see.

**Comment by indytechcook** [comment]
Ok, build is now working

**Comment by vietnogi** [comment]
can we test the build using your branch?  If so, whats steps do we know about?

**Comment by indytechcook** [comment]
gulp serve-dev and gulp serve-build still work.

gulp inject does all of the pre-processing
gulp optimize does the min, concat and image/font optimization and injection along with angular template collecting.

**Comment by indytechcook** [comment]
This also adds a angular constant "apiUrl"

More constants can be added by adding elements to config.ngConstants

File: gulpfile.js [hunk 2]
```diff
@@ -129,33 +147,10 @@ gulp.task('templatecache', ['clean-code'], function () {
...
 });
 
-/**
- * Wire-up the bower dependencies
- * @return {Stream}
- */
-gulp.task('wiredep', ['scss', 'jade'], function () {
-  log('Wiring the bower dependencies into the html');
-
-  var wiredep = require('wiredep').stream;
```
Comment (tladendo) [inline gulpfile.js]: You removed `wiredep` because we're doing all this manually in `index.jade`, right?

Comment (indytechcook) [inline gulpfile.js]: Yes

**Comment by tladendo** [comment]
lgtm

**Comment by tladendo** [comment]
lgtm

**Comment by vietnogi** [comment]
we should minimize the html

**Comment by vietnogi** [comment]
mobile-landscape-xxx.css is not minimized

**Comment by vietnogi** [comment]
How are we going to handle image urls in css files?  Currently, they are relative during development and would break for the build.

**Comment by indytechcook** [comment]
> Minimize html.

Sure, i can add https://github.com/jonschlinkert/gulp-htmlmin.

> mobile-landscape-xxx.css is not minimized

Thanks, that was blank last time around.  I'll get that fixed.

> How are we going to handle image urls in css files? Currently, they are relative during development and would break for the build.

magic!  jk.  I'll look into it unless there is something you have used in the past.  In the past, I've just kept the images in the same relative location but I'm very open to thoughts here.

**Comment by vietnogi** [comment]
We can probably use http://compass-style.org/reference/compass/helpers/urls/ in junction with http://compass-style.org/help/documentation/sass-based-configuration-options/

**Comment by indytechcook** [comment]
html minifing and  all css are minifing.  Next is to add images

**Comment by tladendo** [comment]
what's the word on injecting the mock stuff?

**Comment by indytechcook** [comment]
Build working to S3:  http://ap-work-client-site-dev.s3-website-us-east-1.amazonaws.com/

!!

**Pull request merged**